sdk/keyvault/Azure.Security.KeyVault.Administration/src/KeyVaultBackupClient.cs

```diff
@@ -378,5 +376,20 @@ internal virtual Response<FullBackupDetailsInternal> GetBackupDetails(string job
                 throw;
             }
         }
+
+        private static void ParseFolderName(Uri folderUri, out string containerUriString, out string folderName)
+        {
+            string[] uriSegments = folderUri.Segments;
+            if (uriSegments.Length > 3)
+            {
+                containerUriString = folderUri.AbsoluteUri.Substring(0, folderUri.AbsoluteUri.Length - folderUri.AbsolutePath.Length + uriSegments[1].Length);
```

Comment:
This already created an array on the heap when you used `Segments`, but then you resort to string parsing. Rather than all the extra allocations, you could avoid the conditions and just use a combination of `Substring` and `IndexOf` to find the second "/" (skipping the first), then take everything after than from the `Uri.AbsolutePath`. That would result in a lot less allocations and having to condition the code here.

Reply:
Although this is not particularly perf sensitive code, it's a good point:

```
|               Method |      Mean |    Error |   StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|--------------------- |----------:|---------:|---------:|-------:|------:|------:|----------:|
| SegmentsAndSubstring | 180.29 ns | 3.147 ns | 2.944 ns | 0.0582 |     - |     - |     488 B |
|  SubStringAndIndexOf |  75.70 ns | 1.471 ns | 1.694 ns | 0.0277 |     - |     - |     232 B |
```